### PR TITLE
Add validation for affects with wontreport status

### PR DESCRIPTION
**Pull request by costaconrado**
This PR adds validation for services related products with WONTREPORT resolution.

Closes OSIDB-362.

### CODE REVIEW

**Review by osoukup — CHANGES_REQUESTED** [review]
I see that you extended the original validation a bit which is good but I would rather split it into several validation cases so we report these issues individually.

- one is that **wontreport** resolution can only be assigned to services
- second is that **wontreport** can only be assigned to **low** or **moderate** impact and we should consider first affect impact but when it has NOVALUE check the flaw impact instead
- third is that **wontreport** can only be combined with **affected** affectedness

For the third I would do something more general like https://git.prodsec.redhat.com/devops/sfm2/-/blob/master/sfm2/validation/flaw.py#L2138 (I hope there is a special Jira issue for it).

**Comment by costaconrado** [comment]
I have submitted changes separating case 1 from 2. Case 3 is being handled by OSIDB-360 / PR125.

File: osidb/models.py [hunk 1]
```diff
@@ -1157,6 +1158,39 @@ def delegated_resolution(self):
 
         return Affect.AffectFix.AFFECTED
 
+    def _validate_wontreport_products(self):
+        """
+        Validate that affected/wontreport only can be used for
+        to products associated with services
+        """
+        if (
+            self.affectedness == Affect.AffectAffectedness.AFFECTED
```
Comment (osoukup) [inline osidb/models.py:1167]: I would not check affectedness here at all. This should be really just about WONTREPORT resolution alone.

File: osidb/models.py [hunk 2]
```diff
@@ -1157,6 +1158,39 @@ def delegated_resolution(self):
...
+                    "wontreport can only be associated with service products"
+                )
+
+    def _validate_wontreport_severity(self):
+        """
+        Validate that affected/wontreport only can be used for
+        low or moderate severity flaws
+        """
+        if (
+            self.affectedness == Affect.AffectAffectedness.AFFECTED
```
Comment (osoukup) [inline osidb/models.py:1185]: Same here

**Review by osoukup — CHANGES_REQUESTED** [review]
I have still a small complain :angel:

File: osidb/tests/test_flaw.py [hunk 3]
```diff
@@ -1194,3 +1195,79 @@ def test_flaw_affects_rhscl_invalid_collection(
...
+        ],
+    )
+    def test_validate_wontreport_products(
+        self, impact, affectedness, resolution, prefix, should_raise
+    ):
+        """
+        Tests that only products associated services, having a impact
+        of LOW or MODERATE, can be marked as AFFECTED by a WONTREPORT affect
+        """
+        SERVICES_PRODUCTS.append("service-product")
```
Comment (Elkasitu) [inline osidb/tests/test_flaw.py:1246]: Use monkeypatch [1] instead, if you don't there's a risk that tests defined after this one that depend on `SERVICES_PRODUCTS` either directly or indirectly will use it with `"service-product"` as a valid value which might not be what we want.

[1] https://docs.pytest.org/en/7.1.x/how-to/monkeypatch.html

**Review by Elkasitu — APPROVED** [review]
Just some minor comments, otherwise LGTM

File: osidb/models.py (hunk 1 above)
Comment (costaconrado) [reply osidb/models.py]: submitted a fixture.

**Review by osoukup — APPROVED** [review]
LGTM

**Pull request merged**